service/matching/workers/registry_impl.go

```diff
@@ -179,40 +182,23 @@ func (b *bucket) evictByCapacity(threshold time.Time) bool {
 }
 
 // NewRegistry creates a workers heartbeat registry with the given parameters.
-func NewRegistry(lc fx.Lifecycle, metricsHandler metrics.Handler, enableWorkerPluginMetrics dynamicconfig.BoolPropertyFn) Registry {
-	m := newRegistryImpl(
-		defaultBuckets,
-		defaultEntryTTL,
-		defaultMinEvictAge,
-		defaultMaxEntries,
-		defaultEvictionInterval,
-		metricsHandler,
-		enableWorkerPluginMetrics,
-	)
-
+func NewRegistry(lc fx.Lifecycle, params RegistryParams) Registry {
+	m := newRegistryImpl(params)
 	lc.Append(fx.StartStopHook(m.Start, m.Stop))
```

Comment:
I would arrange this differently so that we keep the fx stuff out of this file and just in fx.go, and either do it without a struct or use fx.In, but I don't think we should hold this up over that, we can clean it up later.